### ISSUE TITLE
Specify how to override the termtype of a fnml:FunctionMap result

### ISSUE THREAD

**Issue by bjdmeest** [kg-construct/rml-fno-spec#7]
A `fnml:FunctionMap` always returns an RDF term [[rdf-concepts]] or list thereof,
thus already has things such as termtype, datatype, etc. defined.
We probably want to allow to override these datatypes within an RML mapping.
If so, do we need to specify how to override these datatypes?
Are there standardized RDF coercion rules?
Is this related to https://www.w3.org/TR/r2rml/#natural-mapping?

**Comment by dachafra** [comment]
I was thinking exactly about the relation between SQL and the mapping in R2RML as you mentioned. Could be a way to solve it following a similar approach

**Comment by bjdmeest** [comment]
https://www.w3.org/TR/json-ld11-api/#data-round-tripping might also be relevant

**Comment by samiscoding** [comment]
Do we already have any specifications in the case of having a list of output values? I think it falls more into the behavior of the engine but still makes sense to have the definition of the correct behavior.

**Comment by samiscoding** [comment]
Already got my answer! :D It's still an open issue in mapping-challenges. I think makes sense to specifically have an issue on that for rml-fno

**Comment by andimou** [comment]
just to be on the same page, you mean this challeng @samiscoding right?

https://github.com/kg-construct/mapping-challenges/tree/main/challenges/multivalue-references

**Comment by samiscoding** [comment]
> just to be on the same page, you mean this challeng @samiscoding right?
> 
> https://github.com/kg-construct/mapping-challenges/tree/main/challenges/multivalue-references

Yes, but more specifically [this challenge] (https://github.com/kg-construct/mapping-challenges/tree/main/challenges/process-multivalue-reference)
